File: backend/Models/Permiso.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SistemaGestionDocumental.Models;

[Table("permisos")]
public class Permiso
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Required]
    [Column("codigo")]
    [StringLength(50)]
    public string Codigo { get; set; } = string.Empty;

    [Required]
    [Column("nombre")]
    [StringLength(100)]
    public string Nombre { get; set; } = string.Empty;

    [Column("descripcion")]
    [StringLength(500)]
    public string? Descripcion { get; set; }

    [Column("modulo")]
    [StringLength(50)]
    public string Modulo { get; set; } = "Documentos";

    [Column("activo")]
    public bool Activo { get; set; } = true;

    // Relación muchos a muchos con roles
    public virtual ICollection<RolPermiso> RolPermisos { get; set; } = new List<RolPermiso>();
}

File: frontend/lib/models/permiso.dart
class Permiso {
  final int id;
  final String codigo;
  final String nombre;
  final String? descripcion;
  final String modulo;
  final bool activo;

  Permiso({
    required this.id,
    required this.codigo,
    required this.nombre,
    this.descripcion,
    required this.modulo,
    required this.activo,
  });

  factory Permiso.fromJson(Map<String, dynamic> json) {
    return Permiso(
      id: json['id'] as int,
      codigo: json['codigo'] as String,
      nombre: json['nombre'] as String,
      descripcion: json['descripcion'] as String?,
      modulo: json['modulo'] as String,
      activo: json['activo'] as bool,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'codigo': codigo,
      'nombre': nombre,
      'descripcion': descripcion,
      'modulo': modulo,
      'activo': activo,
    };
  }
}
